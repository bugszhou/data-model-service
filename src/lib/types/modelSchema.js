import { typeEnum } from './modelConf';

export default {
  type: {
    type: 'string',
    description: '类型',
    oneOf: typeEnum,
    required: true,
  },
  description: {
    type: 'string',
    description: '描述',
    required: false,
  },
  from: {
    type: 'string',
    description: '数据来源',
    required: true,
  },
  properties: {
    type: 'object',
    description: '属性',
    required: false,
  },
  default: {
    type: 'function',
    description: '默认值',
    required: false,
  },
  required: {
    type: 'boolean',
    description: '是否必传',
    required: false,
  },
};