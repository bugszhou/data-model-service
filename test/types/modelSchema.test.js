import modelSchema from '../../src/lib/types/modelSchema';

describe('model的schema测试', () => {
  it('model schema快照', () => {
    expect(modelSchema).toMatchSnapshot();
  });
});

