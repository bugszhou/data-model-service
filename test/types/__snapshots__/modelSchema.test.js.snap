// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`model的schema测试 model schema快照 1`] = `
Object {
  "default": Object {
    "description": "默认值",
    "required": false,
    "type": "function",
  },
  "description": Object {
    "description": "描述",
    "required": false,
    "type": "string",
  },
  "from": Object {
    "description": "数据来源",
    "required": true,
    "type": "string",
  },
  "properties": Object {
    "description": "属性",
    "required": false,
    "type": "object",
  },
  "required": Object {
    "description": "是否必传",
    "required": false,
    "type": "boolean",
  },
  "type": Object {
    "description": "类型",
    "oneOf": Array [
      "string",
      "number",
      "integer",
      "boolean",
      "object",
      "array",
      "null",
    ],
    "required": true,
    "type": "string",
  },
}
`;
